### PR TITLE
ci/release: Use GCP to up/download signed binaries

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .azure-pipelines/stage/publish.yml [hunk 1]
```diff
@@ -402,18 +404,24 @@ jobs:
...
+      - bash: |
+          set -e
+          if [[ "$BUILD_REASON" == "PullRequest" ]]; then
+              DOWNLOAD_PATH="$(git rev-parse HEAD | head -c7)"
+          else
+              DOWNLOAD_PATH="${SYSTEM_PULLREQUEST_PULLREQUESTNUMBER:-${BUILD_SOURCEBRANCHNAME}}"
+          fi
+          curl -sLO "https://storage.googleapis.com/${{ parameters.bucketGCP }}/${DOWNLOAD_PATH}/release/release.signed.tar.zst"
+          mkdir -p $(Build.StagingDirectory)/release.signed
+          cp -a release.signed.tar.zst $(Build.StagingDirectory)/release.signed
```
Comment (wbpcode) [inline .azure-pipelines/stage/publish.yml:423]: nit: will move be better here?

**Review by wbpcode — DISMISSED** [review]
LGTM overall, only one nit comment.

File: .azure-pipelines/stage/publish.yml (hunk 1 above)
Comment (phlax) [reply .azure-pipelines/stage/publish.yml]: not sure it matters that much, but updated anyway

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**